ExampleMod/Content/Projectiles/ExamplePaperAirplaneProjectile.cs

```diff
@@ -0,0 +1,166 @@
+				Projectile.velocity.X = MathHelper.Clamp(Projectile.velocity.X + Main.WindForVisuals * 0.006f, -6f, 6f);
+
+				// Switch direction when the current velocity and the oldVelocity have different signs.
+				if (Projectile.velocity.X * Projectile.oldVelocity.X < 0f) {
+					Projectile.direction *= -1; // Reverse the direction
+					Projectile.ai[0] = -Main.rand.Next(120, 300); // Set Projectile.ai[0] to a random number from -120 to -599.
+					Projectile.netUpdate = true; // Sync the projectile in a multiplayer game.
+				}
+			}
+
+			// Set the rotation and spriteDirection
+			Projectile.rotation = Projectile.velocity.ToRotation();
+			Projectile.spriteDirection = Projectile.direction;
+
+			// Let's add some dust for special effect. In this case, it runs every other tick (30 ticks per second).
+			if (Projectile.timeLeft % 2 == 0) {
+				Dust.NewDustPerfect(new Vector2(Projectile.Center.X - (Projectile.width * Projectile.direction), Projectile.Center.Y), ModContent.DustType<Dusts.Sparkle>(), null, 0, default, 0.5f); //Here we spawn the dust at the back of the projectile with half scale.
+			}
+		}
+
+		// We need to do draw the projectile manually. If you don't include this, the projectile will be facing the wrong direction when flying left.
+		public override bool PreDraw(ref Color lightColor) {
+			// This is where we specify which way to flip the sprite. If the projectile is moving to the left, then flip it vertically.
+			SpriteEffects spriteEffects = ((Projectile.spriteDirection <= 0) ? SpriteEffects.FlipVertically : SpriteEffects.None);
+
+			// Getting texture of projectile
+			Texture2D texture = (Texture2D)ModContent.Request<Texture2D>(Texture);
+
+			// Get this frame on texture. If you want your projectile to be animated, see ExampleAdvancedAnimatedProjectile
+			Rectangle sourceRectangle = new Rectangle(0, 0, texture.Width, texture.Height);
+
+			Vector2 origin = sourceRectangle.Size() / 2f;
+
+			// Applying lighting and draw our projectile
+			Color drawColor = Projectile.GetAlpha(lightColor);
+			Main.EntitySpriteDraw(texture,
+				Projectile.Center - Main.screenPosition + new Vector2(0f, Projectile.gfxOffY),
+				sourceRectangle, drawColor, Projectile.rotation, origin, Projectile.scale, spriteEffects, 0);
+
+			// It's important to return false, otherwise we also draw the original texture.
+			return false;
+		}
+
+		public override void Kill(int timeLeft) {
+			SoundEngine.PlaySound(SoundID.Item10, Projectile.position); // Play a sound when the projectile dies. In this case, that is when it hits a block or a liquid.
+
+			if (Projectile.owner == Main.myPlayer && !Projectile.noDropItem) {
+				int dropItemType = ModContent.ItemType<Items.ExamplePaperAirplane>(); // This the item we want the paper airplane to drop.
+				int newItem = Item.NewItem(Projectile.GetSource_DropAsItem(), (int)Projectile.position.X, (int)Projectile.position.Y, Projectile.width, Projectile.height, dropItemType); // Create a new item in the world.
```

Comment:
```suggestion
				int newItem = Item.NewItem(Projectile.GetSource_DropAsItem(), Projectile.getRect(), dropItemType); // Create a new item in the world.
```
Alternatively, `Projectile.Hitbox`